### PR TITLE
SDIT-1710 Check Agency exists before determining access

### PR DESCRIPTION
**Pull request by karenmillermoj**
This mimics the check for bookings

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java [hunk 1]
```diff
@@ -214,7 +215,7 @@ public Set<String> getAgencyIds(boolean allowInactive) {
      * @throws AccessDeniedException   if current user does not have access to this agency and accessDeniedError is true.
      */
     public void verifyAgencyAccess(final String agencyId, boolean accessDeniedError, boolean allowInactive) {
-        Objects.requireNonNull(agencyId, "agencyId is a required parameter");
+        checkAgencyExists(agencyId, allowInactive ? ALL: ACTIVE_ONLY);
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:218]: Not sure about this - does it matter if a 403 is returned when the agency doesnt exist? Just thinking about the extra db calls 🤷

**Review by steverendell — APPROVED** [review]
lgtm

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -543,15 +543,20 @@ private void validateScheduledEventsRequest(final LocalDate fromDate, final Loca
...
+            return;
+        }
 
         final var agencyIds = agencyService.getAgencyIds(false);
         if (AuthenticationFacade.hasRoles("INACTIVE_BOOKINGS")) {
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
 
         if (agencyIds.isEmpty()) {
+            checkBookingExists(bookingId);
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:559]: All of the "if" blocks are starting to look a bit unreadable.  This should hopefully start to tidy up once we can remove the checks for different types of exceptions to throw and also we will not need to do explicit client only checks

**Review by steverendell — APPROVED** [review]
LGTM!

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java (hunk 1 above)
Comment (karenmillermoj) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java]: As discussed - move checks into non-happy path

**Pull request merged**